sdk/keyvault/azure-security-keyvault-administration/README.md

```diff
@@ -252,87 +291,118 @@ The following sections provide several code snippets covering some of the most c
 > Note : You should add `System.in.read()` or `Thread.sleep()` after the function calls in the main class/thread to allow async functions/operations to execute and finish before the main application/thread exits.
 
 ### List role definitions asynchronously
-List the role definitions in the key vault by calling `listRoleDefinitions`.
+List the role definitions in the key vault by calling `listRoleDefinitions()`.
 
+<!-- embedme src/samples/java/ReadmeSamples.java#141-143 -->
 ```java
-KeyVaultRoleAssignmentScope roleAssignmentScope = KeyVaultRoleAssignmentScope.GLOBAL;
+keyVaultAccessControlAsyncClient.listRoleDefinitions(KeyVaultRoleScope.GLOBAL)
+    .subscribe(roleDefinition ->
+        System.out.printf("Retrieved role definition with name '%s'.%n", roleDefinition.getName()));
+```
+
+### Create or update a role definition asynchronously
+Create or update a role definition in the key vault. The following example shows how to create a role definition with a randomly generated name.
 
-accessControlAsyncClient.listRoleDefinitions(roleAssignmentScope))
+<!-- embedme src/samples/java/ReadmeSamples.java#150-153 -->
+```java
+keyVaultAccessControlAsyncClient.setRoleDefinition(KeyVaultRoleScope.GLOBAL)
     .subscribe(roleDefinition ->
-        System.out.printf("Retrieved role definition with name \"%s\" and type \"%s\"%n", roleDefinition.getName(),
-                roleDefinition.getType()));
+        System.out.printf("Created role definition with randomly generated name '%s' and role name '%s'.%n",
+            roleDefinition.getName(), roleDefinition.getRoleName()));
 ```
 
-### List role assignments asynchronously
-List the role assignments in the key vault by calling `listRoleAssignments`.
+### Retrieve a role definition asynchronously
+Get an existing role definition. To do this, the 'name' property from an existing role definition is required.
 
+<!-- embedme src/samples/java/ReadmeSamples.java#160-165 -->
 ```java
-KeyVaultRoleAssignmentScope roleAssignmentScope = KeyVaultRoleAssignmentScope.GLOBAL;
+String roleDefinitionName = "<role-definition-name>";
+
+keyVaultAccessControlAsyncClient.getRoleDefinition(KeyVaultRoleScope.GLOBAL, roleDefinitionName)
+    .subscribe(roleDefinition ->
+        System.out.printf("Retrieved role definition with name '%s' and role name '%s'.%n",
+            roleDefinition.getName(), roleDefinition.getRoleName()));
+```
+
+### Delete a role definition asynchronously
+Delete a role definition. To do this, the 'name' property from an existing role definition is required.
 
-accessControlAsyncClient.listRoleAssignments(roleAssignmentScope))
+<!-- embedme src/samples/java/ReadmeSamples.java#172-175 -->
```

Comment:
The path to `ReadmeSamples.java` seems to be incorrect (package name should be included in the path).